server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/DateHistogramAggregationBuilder.java

```diff
@@ -417,9 +419,27 @@ protected ValuesSourceAggregatorFactory innerBuild(
         AggregatorFactory parent,
         AggregatorFactories.Builder subFactoriesBuilder
     ) throws IOException {
-        DateHistogramAggregationSupplier aggregatorSupplier = context.getValuesSourceRegistry().getAggregator(REGISTRY_KEY, config);
+        IndexMode indexMode = context.getIndexSettings().getMode();
+        final DateIntervalWrapper.IntervalTypeEnum dateHistogramIntervalType = dateHistogramInterval.getIntervalType();
+        if (context.getIndexSettings().getMode().validateCalendarIntervalType(dateHistogramIntervalType)) {
```

Comment:
Can this method be `void` and throw for invalid values? Could you do the same for the time zone below?

Reply:
Yes I was doing that...returning true/false is confusing...is it returning false if validation is ok or not?

Reply:
I'd just validate and throw.